src/server.c

```diff
@@ -6338,6 +6383,7 @@ int main(int argc, char **argv) {
         serverLog(LL_WARNING,"Server initialized");
     #ifdef __linux__
         linuxMemoryWarnings();
+        server.thp_enabled = THPIsEnabled();
```

Comment:
```suggestion
        server.thp_enabled = THPIsEnabled() && !server.disable_thp;
```
`Thp` may be disabled.

Reply:
before this line, if server.disable_THP, we will try to disable THP (and we may fail to do that), so here we only check again i think

Reply:
I have tested, if `THPIsEnabled()` is true, even if `prctl(PR_SET_THP_DISABLE, 1, 0, 0, 0)` is successful, `THPIsEnabled()` is still true.
Maybe this line can be moved to `linuxMemoryWarnings`.

Reply:
Thanks @sundb , i had second look, i find `prctl` only take effect on calling thread instead of entire OS, so the result of `THPIsEnabled` still is not changed. But i have question, we released the pages that belong to child process, does child process take effect on child process after calling `prctl` by parents?